### PR TITLE
[SDK-1924] Only apply DFPPA to DFPPA-enabled endpoints

### PR DESCRIPTION
**Pull request by nidal-stytch**
[[iOS] Update DFPPA to only apply to DFPPA-enabled endpoints](https://linear.app/stytch/issue/SDK-1924/[ios]-update-dfppa-to-only-apply-to-dfppa-enabled-endpoints)

## Changes:

1. Modify all the methods in `NetworkingRouter` to accept the parameter `useDFPPA: Bool` that has a default value of false.
2. Modify the logic in `NetworkingClient+Live` to make sure `dfpEnabled == true && useDFPPA == true` before preforming any DFP operations.
3. Modify the `Error` extension, now named `Error+StytchErrorInfo` to include additional error types and fields.

## Checklist:
- [x] I have verified that this change works in the relevant demo app, or N/A
- [x] I have added or updated any tests relevant to this change, or N/A
- [x] I have updated any relevant README files for this change, or N/A

### CODE REVIEW

File: Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift [hunk 1]
```diff
@@ -14,22 +14,23 @@ public extension StytchClient {
         // sourcery: AsyncVariants, (NOTE: - must use /// doc comment styling)
         /// Wraps Stytch's OTP [sms/login_or_create](https://stytch.com/docs/api/log-in-or-create-user-by-sms), [whatsapp/login_or_create](https://stytch.com/docs/api/whatsapp-login-or-create), and [email/login_or_create](https://stytch.com/docs/api/log-in-or-create-user-by-email-otp) endpoints. Requests a one-time passcode for a user to log in or create an account depending on the presence and/or status current account.
         public func loginOrCreate(parameters: Parameters) async throws -> OTPResponse {
-            try await router.post(to: .loginOrCreate(parameters.deliveryMethod), parameters: parameters)
+            try await router.post(to: .loginOrCreate(parameters.deliveryMethod), parameters: parameters, useDFPPA: true)
```
Comment (jhaven-stytch) [inline Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift:17]: For `loginOrCreate` and `send` OTP methods, it's correct for SMS/WhatsApp, but Email is not DFPPA protected, so will need to switch on the delivery method

**Review by jhaven-stytch — CHANGES_REQUESTED** [review]
Just the one comment about consumer OTPs

File: Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift (hunk 1 above)
Comment (nidal-stytch) [reply Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift]: Ok fixed!

File: Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift [hunk 2]
```diff
@@ -134,5 +139,16 @@ public extension StytchClient.OTP {
...
+        var shouldUseDFPPA: Bool {
+            switch self {
+            case .sms(phoneNumber: _, enableAutofill: _):
+                return true
+            case .whatsapp(phoneNumber: _):
+                return true
+            case .email(email: _, loginTemplateId: _, signupTemplateId: _):
+                return false
+            }
+        }
```
Comment (nidal-stytch) [inline Sources/StytchCore/StytchClient/OTP/SytchClient+OTP.swift:152]: @jhaven-stytch fixed here!

**Review by jhaven-stytch — APPROVED** [review]

**Pull request merged**